src/Events/Span.php

```diff
@@ -82,6 +82,10 @@ public function start(float $startTime = null)
         }
 
         $this->timer = $this->timerFactory->newTimer($startTime);
+        // Start the timer only if there was not a start time
+        if (null === $startTime) {
+            $this->timer->start();
+        }
         $this->timestamp = (int) round($this->timer->getStartTime() * 1000000);
```

Comment:
Why do we need the timestamp here if we already have inside the timer? Seems to me that `Span` can get rid of timestamp member in favor of `Timer`.

Reply:
The parent `EventBean` defines the timestamp. It's set to the timestamp of the class creation, though, which is not necessarily the correct value when dealing with a `Transaction` or a `Span` which has its own `Timer`. While I suspect the only use of the timestamp is when the object is converted to JSON, I have not fully evaluated that assumption. I felt it was safer to make the minimum change to the current implementation.

At the same time, I'm trying to increase test coverage so we can make more substantial changes in the future with a higher level of confidence.

Reply:
Agreed. I think there is na opportunity to refactor some code in `Span`. Lets discuss latter.